ga4gh/backend.py

```diff
@@ -141,6 +161,37 @@ def startProfile(self):
     def endProfile(self):
         pass
 
+    def callSetsGenerator(self, request):
+        """
+        Returns a generator over the (callSet, nextPageToken) pairs defined by
+        the specified request.
+        """
+        # if no variantSetIds are input from client,
+        # then set variantSetIds to all variantSetIds
+        if request.variantSetIds == []:
+            variantSetIds = self._variantSetIds
+        else:
+            variantSetIds = request.variantSetIds
+        name = request.name
+        if request.pageToken is not None:
+            startVariantSetIndex, startCallSetPosition = self.parsePageToken(
+                request.pageToken, 2)
+        else:
+            startVariantSetIndex = 0
+            startCallSetPosition = 0
+
+        for variantSetIndex in range(startVariantSetIndex, len(variantSetIds)):
+            variantSetId = variantSetIds[variantSetIndex]
+            if variantSetId in self._variantSetIdMap:
+                variantSet = self._variantSetIdMap[variantSetId]
+                callSetIterator = variantSet.getCallSets(
+                    name, startCallSetPosition)
+                for callSet, callSetPosition in callSetIterator:
+                    callSet.variantSetIds = self._callSetIdMap[callSet.id]
+                    nextPageToken = "{0}:{1}".format(
```

Comment:
We return an empty page of results at the end here. We need to return `None` as the `nextPageToken` when we are at the last result.